### ISSUE TITLE
TwinCAT POU Pre/Post Newlines Hook

### ISSUE THREAD

**Issue by ZLLentz** [pcdshub/pre-commit-hooks#10]
I'm considering making a pre-commit hook that inserts newlines at the start and end of TwinCAT POUs (and some of the other files?). When a TwinCAT POU starts and ends with a newline, it is much easier to review in github. Snippit from a current PR of mine that does not do this:
```
    <Implementation>
      <ST><![CDATA[fbNCErrorFFO(
    stMotionStage := stMotionStage,
    fbFFHWO := fbFFHWO,
    bReset := bReset,
    sName := sName,
    bTripped => bTripped);]]></ST>
    </Implementation>
```
With pre and post newlines:
```
    <Implementation>
      <ST><![CDATA[
fbNCErrorFFO(
    stMotionStage := stMotionStage,
    fbFFHWO := fbFFHWO,
    bReset := bReset,
    sName := sName,
    bTripped => bTripped);
]]></ST>
    </Implementation>
```
With two pre and two post newlines:
```
    <Implementation>
      <ST><![CDATA[

fbNCErrorFFO(
    stMotionStage := stMotionStage,
    fbFFHWO := fbFFHWO,
    bReset := bReset,
    sName := sName,
    bTripped => bTripped);

]]></ST>
    </Implementation>
```

**Comment by klauer** [comment]
Seems reasonable to me 👍 (DUT, GVL, POU is our normal set of "source files" I believe)

**Comment by ZLLentz** [comment]
Forgot to add, the idea for having these newlines at all comes from some old @slacAWallace PR or comment on a PR of some kind. It's a pain to maintain manually, but as an automatic hook...

**Comment by ZLLentz** [comment]
closed by #23
